src/test/java/org/junitpioneer/jupiter/RetryingTestExtensionTests.java

```diff
@@ -226,6 +292,17 @@ void skipByAssumption() {
 			Assumptions.assumeFalse(true);
 		}
 
+		@RetryingTest(value = 3, onExceptions = IllegalArgumentException.class)
+		void failsFirstWithExpectedExceptionThenSkippedByAssumption() {
+			EXECUTION_COUNT++;
+			if (EXECUTION_COUNT == 1) {
+				throw new IllegalArgumentException();
+			}
+			if (EXECUTION_COUNT == 2) {
+				Assumptions.assumeFalse(true);
```

Comment:
Since we refer to this specific exception in our code and docs?

```suggestion
				throw new TestAbortedException();
```